tests/test_tags.py

```diff
@@ -1364,6 +1364,44 @@ def test_cpython_first_none_any_tag(self, monkeypatch):
         interpreter = f"cp{tags.interpreter_version()}"
         assert tag == tags.Tag(interpreter, "none", "any")
 
+    def test_emscripten(self, mock_interpreter_name, monkeypatch):
+        expected_interpreter = "cp" + tags._version_nodot(sys.version_info[:2])
+        if mock_interpreter_name("CPython"):
+            monkeypatch.setattr(
+                tags, "_cpython_abis", lambda _1, _2: [expected_interpreter]
+            )
+        monkeypatch.setattr(platform, "system", lambda: "Emscripten")
+        monkeypatch.setattr(
+            sysconfig, "get_platform", lambda: "emscripten-3.1.58-wasm32"
+        )
+        assert list(tags.platform_tags()) == ["emscripten_3_1_58_wasm32"]
+        result = list(tags.sys_tags())
+        assert result[0] == tags.Tag(
+            expected_interpreter, "cp312", "emscripten_3_1_58_wasm32"
```

Comment:
Should the cp12 here also be adjusted based on the current Python version?

Reply:
I think it would make sense to compute the Python version dynamically so that it doesn't need updating afterwards – maybe by reading the value of `pyodide config get python_version` (but I don't know if that would be available), or by adding a mapping of Python versions and emscripten versions. Though I think recognising that "pyodide_<...>" and "emscripten_<...>" in the platform tags is the more important bit.

Reply:
I was wondering because it looks like the only place where `12` is hardcoded in the test, and in the last CI run only the 3.12 runs succeeded, so maybe changing the assert could fix CI?

Reply:
Yes, at least for fixing the test failure, it should address the problem on an immediate basis. :)